pages/docs/reference/collections-overview.md

```diff
@@ -0,0 +1,269 @@
+
+Let's walk through the interfaces and their implementations.
+
+### Collection
+
+`Collection<T>` is the root of the collection hierarchy. This interface represents the common behavior of a read-only collection: retrieving size, checking item membership, and so on.
+`Collection` inherits from the `Iterable<T>` interface that defines the operations for iterating elements. You can use `Collection` as a parameter of a function that applies to different collection types. For more specific cases, use the `Collection`'s inheritors: `Set` and `List`.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun printAll(strings: Collection<String>) {
+        for(s in strings) print("$s ")
+        println()
+    }
+    
+fun main() {
+    val stringList = listOf("one", "two", "one")
+    printAll(stringList)
+    
+    val stringSet = setOf("one", "two", "three")
+    printAll(stringSet)
+}
+```
+</div>
+
+`MutableCollection` is a `Collection` with write operations, such as add and remove.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun List<String>.getShortWordsTo(shortWords: MutableList<String>, maxLength: Int) {
+    this.filterTo(shortWords) { it.length <= maxLength}
+    // throwing away the articles
+    val articles = setOf("a", "A", "an", "An", "the", "The")
+    shortWords -= articles
+}
+
+fun main() {
+    val words = "A long time ago in a galaxy far far away".split(" ")
+    val shortWords = mutableListOf<String>()
+    words.getShortWordsTo(shortWords, 3)
+    println(shortWords)
+}
+
+```
+</div>
+
+### List
+
+`List<T>` stores elements in a specified order and provides indexed access to them. Indices start from zero – the index of the first element – and go to `lastIndex` which is the `(list size - 1)`. 
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+//sampleStart
+    val numbers  = listOf("one", "two", "three", "four")
+    println("Number of elements: ${numbers.size}")
+    println("Third element: ${numbers.get(2)}")
+    println("Fourth element: ${numbers[3]}")
+    println("Index of element \"two\" ${numbers.indexOf("two")}")
+//sampleEnd
+}
+```
+</div>
+
+List elements (including nulls) can duplicate: a list can contain any number of equal objects or occurrences of a single object.
+Two lists are considered equal if they have the same sizes and [structurally equal](equality.html#structural-equality) elements at the same positions. 
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+data class Person(var name: String, var age: Int)
+
+fun main() {
+//sampleStart
+    val bob = Person("Bob", 31)
+    val people = setOf<Person>(Person("Adam", 20), bob, bob)
+    val people2 = setOf<Person>(Person("Adam", 20), Person("Bob", 31), bob)
+    println(people.equals(people2))
+    bob.age = 32
+    println(people.equals(people2))
```

Comment:
Use `==` instead of `equals` in this example and `listOf` instead of `setOf`

Reply:
fixed